src/handlers/merge_conflict.rs

```diff
@@ -0,0 +1,330 @@
+    db::issue_data::IssueData,
+    db::PooledClient,
+    github::{
+        Event, GithubClient, Issue, IssuesAction, IssuesEvent, Label, MergeConflictInfo,
+        MergeableState, PushEvent, ReportedContentClassifiers, Repository,
+    },
+    handlers::Context,
+};
+use anyhow::Context as _;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use std::time::Duration;
+use tokio_postgres::Client as DbClient;
+use tracing as log;
+
+/// Key for the database.
+const MERGE_CONFLICT_KEY: &str = "merge-conflict";
+
+/// The amount of time to wait before scanning an unknown mergeable status.
+///
+/// GitHub has a background job which updates the mergeable status. We have to
+/// wait for it to be finished. Unfortunately there is no notification when it
+/// is done. It seems to usually run pretty quickly, but this timeout is set
+/// to be a little conservative just in case it takes a while to compute. I
+/// don't particularly want to loop to avoid hitting GitHub too hard, and this
+/// conflict notification is not that important to be perfect. If it is too
+/// unreliable, then we could add a loop that will try one or two more times.
+const UNKNOWN_RESCAN_DELAY: Duration = Duration::new(60, 0);
+
+/// State stored in the database for a PR.
+#[derive(Debug, Default, Deserialize, Serialize)]
+struct MergeConflictState {
+    /// The GraphQL ID of the most recent warning comment.
+    ///
+    /// After the conflict is resolved, this will be set to `None`.
+    last_warned_comment: Option<String>,
+}
+
+pub(super) async fn handle(
+    ctx: &Context,
+    event: &Event,
+    config: &MergeConflictConfig,
+) -> anyhow::Result<()> {
+    match event {
+        Event::Push(push) => handle_branch_push(ctx, config, push).await,
+        Event::Issue(IssuesEvent {
+            action: IssuesAction::Opened | IssuesAction::Reopened | IssuesAction::Synchronize,
+            repository,
+            issue,
+            ..
+        }) if issue.pull_request.is_some() => {
+            handle_pr(ctx, config, repository.clone(), issue).await
+        }
+        _ => Ok(()),
+    }
+}
+
+/// Handles a push to a branch in the repository.
+///
+/// This will scan open PRs to see if any of them are now unmergeable.
+async fn handle_branch_push(
+    ctx: &Context,
+    config: &MergeConflictConfig,
+    push: &PushEvent,
+) -> anyhow::Result<()> {
+    let git_ref = push.git_ref.clone();
+    let Some(branch_name) = push.git_ref.strip_prefix("refs/heads/") else {
+        log::trace!("ignoring push to {git_ref}");
+        return Ok(());
+    };
+    let branch_name = branch_name.to_string();
+    let push_sha = push.after.to_string();
+    let config = config.clone();
+    let repo = push.repository.clone();
+    let db = ctx.db.get().await;
+    // Spawn since this can trigger a lot of work.
+    tokio::task::spawn(async move {
+        // See module note about locking.
+        let gh = GithubClient::new_from_env();
```

Comment:
Is there a reason we need a new client vs. using the existing ctx.github client? Seems like that should be Cloneable?

Reply:
Yea, that sounds good! For some reason I was just assuming it wasn't cloneable (or maybe I was concerned about how the connection pooling would work?).